### PR TITLE
Support XPU DDP training and autocast for LowBitMatmul

### PR DESCRIPTION
**Pull request by yangw1234**
## Description

1. Support XPU DDP training by patching transformers
2. Add autocast support for LowBitMatmul to speedup mixed precision training

### 1. Why the change?

Faster end to end training time using multiple intel GPUs

### 2. User API changes

None

### 4. How to test?
Manually tested using 4 PVC 1100 on alpaca dataset

### CODE REVIEW

File: python/llm/src/bigdl/llm/transformers/xpu_customize_fwd.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+except ModuleNotFoundError:
+    np = None  # type: ignore[assignment]
+from typing import Any
+
+
+def _cast(value, dtype):
+    if isinstance(value, torch.Tensor):
+        is_eligible = (
+            value.is_floating_point()
+            and value.is_cuda
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/xpu_customize_fwd.py:115]: why `is_cuda`?

**Review by jason-dai — APPROVED** [review]
LGTM

File: python/llm/src/bigdl/llm/transformers/xpu_customize_fwd.py (hunk 1 above)
Comment (yangw1234) [reply python/llm/src/bigdl/llm/transformers/xpu_customize_fwd.py]: fixed

File: python/llm/src/bigdl/llm/transformers/low_bit_linear.py [hunk 2]
```diff
@@ -378,8 +385,7 @@ def forward(self, x: torch.Tensor):
                 result = result.view(new_shape)
                 if self.bias is not None:
                     result += self.bias
-
-        return result.to(x.dtype)
+        return result
```
Comment (hkvision) [inline python/llm/src/bigdl/llm/transformers/low_bit_linear.py:388]: This line of change causes this issue: https://github.com/analytics-zoo/nano/issues/639
@yangw1234 Please take a look.

**Pull request merged**